src/pages/workspace/WorkspacesListRow.tsx

```diff
@@ -0,0 +1,168 @@
+import React, {useMemo} from 'react';
+import {View} from 'react-native';
+import {SvgProps} from 'react-native-svg';
+import Avatar from '@components/Avatar';
+import Icon from '@components/Icon';
+import * as Illustrations from '@components/Icon/Illustrations';
+import Text from '@components/Text';
+import ThreeDotsMenu from '@components/ThreeDotsMenu';
+import withCurrentUserPersonalDetails, {WithCurrentUserPersonalDetailsProps} from '@components/withCurrentUserPersonalDetails';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as PersonalDetailsUtils from '@libs/PersonalDetailsUtils';
+import {AvatarSource} from '@libs/UserUtils';
+import CONST from '@src/CONST';
+import {Policy} from '@src/types/onyx';
+
+type WorkspacesListRowProps = WithCurrentUserPersonalDetailsProps & {
+    /** Name of the workspace */
+    title: Policy['name'];
+
+    /** Account ID of the workspace's owner */
+    ownerAccountID: Policy['ownerAccountID'];
+
+    /** Type of workspace. Type personal is not valid in this context so it's omitted */
+    workspaceType: typeof CONST.POLICY.TYPE.FREE | typeof CONST.POLICY.TYPE.CORPORATE | typeof CONST.POLICY.TYPE.TEAM;
+
+    /** Icon to show next to the workspace name */
+    workspaceIcon?: AvatarSource | undefined;
+
+    /** Icon to be used when workspaceIcon is not present */
+    fallbackWorkspaceIcon: AvatarSource | undefined;
+
+    /** Renders the component using big screen layout or small screen layout */
+    isWide: boolean | undefined;
+};
+
+const workspaceTypeIcon = (workspaceType: WorkspacesListRowProps['workspaceType']): React.FC<SvgProps> => {
+    switch (workspaceType) {
+        case CONST.POLICY.TYPE.FREE:
+            return Illustrations.BigRocket;
+        case CONST.POLICY.TYPE.CORPORATE:
+            return Illustrations.HotDogStand;
+        case CONST.POLICY.TYPE.TEAM:
+            return Illustrations.SmallRocket;
+        default:
+            throw new Error(`Don't know which icon to serve for workspace type`);
+    }
+};
+
+function WorkspacesListRow({title, workspaceIcon, fallbackWorkspaceIcon, ownerAccountID, workspaceType, currentUserPersonalDetails, isWide}: WorkspacesListRowProps) {
```

Comment:
Instead of passing a fallback icon as a prop, can we handle fallback in the component above? We can accept a single icon prop for `WorkspaceListRow`.

Reply:
@hayata-suenaga I'm doing this because then I'm passing both those icons to `<Avatar/>`:
```
                <Avatar
                    imageStyles={[styles.alignSelfCenter]}
                    size={CONST.AVATAR_SIZE.DEFAULT}
                    source={workspaceIcon}
                    fallbackIcon={fallbackWorkspaceIcon}
                    name={title}
                    type={CONST.ICON_TYPE_WORKSPACE}
                />
```

@aimane-chnaif any preference if I should go for @hayata-suenaga suggestion or leave it as it is?

Reply:
Ah, I didn't check the interface of the Avatar component. Let's leave it as is.